osquery/filesystem/filesystem.cpp

```diff
@@ -310,7 +309,7 @@ static bool checkForLoops(std::set<int>& dsym_inos, std::string path) {
   if (dsym_inos.find(d_stat.st_ino) == dsym_inos.end()) {
     dsym_inos.insert(d_stat.st_ino);
   } else {
-    VLOG(1) << "Symlink loop detected. Ignoring: " << path;
+    // VLOG(1) << "Symlink loop detected. Ignoring: " << path;
```

Comment:
Why is this commented out?

Reply:
we can't safely log from this function because apparently glog shares the logging context, and it might get executed from multiple threads leading to a deadlock.  This particular message used to be written out to the info-level logs and was making a bunch of noise on normal systems.  It got moved back to verbose logging, but I'd argue that this shouldn't be logged at all and should just be silently skipped.

Reply:
The multiple thread execution is safe, as long as it happens in the same process; what's not safe is that to provide the namespace access we have to fork, and osquery might be in any state and if it so happen that another thread was logging and acquired locks, then these locks are held indefinitely in the forked process, which is now single thread, so any next log call would deadlock.